### PR TITLE
feat(frontend): bundle dependency compatibility view

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
## Summary

Adds a **bundle dependency compatibility view** to the Dependencies tab. It compares the bundle you're viewing (the **candidate** — what would ship over-the-air) against a **baseline** bundle you pick, and tells you whether the update is OTA-safe or needs an app-store update — the same verdict as the CLI `capgo bundle compatibility`.

Builds on the compare-picker fix already merged in #2373 (which gated the picker on `native_packages`). This PR is a single clean commit on top of `main`.

## What it does

**Status-aware dependency diff** — every native package shown with a colour-coded status and version(s):

| Status | Colour | Display |
|---|---|---|
| changed | blue | `old → new` |
| added | green | `New · <version>` |
| removed | red | `Removed · <version>` |
| unchanged | gray | `Unchanged · <version>` |

Counts row: Changed / Added / Removed / Unchanged.

**Compatibility verdict banner** — ✅ Compatible ("can be delivered over-the-air to devices running {baseline}") or ❌ Not compatible ("N packages require an app store update"), with per-package reasons. Logic is ported from the CLI's `getCompatibilityDetails`: version-range intersection (`@std/semver`) plus iOS/Android native checksum changes; a **new** native plugin blocks OTA, a **removed** one is safe.

**Deep-linkable** — selecting a baseline reflects in the URL as `?compare=<versionId>` (history-friendly `router.replace`); loading such a URL pre-selects the baseline.

**Picker hygiene** — the compare dropdown excludes soft-deleted bundles, and falls back past deleted deployments when choosing preferred (deploy-history) baselines, so a channel whose latest prior deploy was deleted still gets a usable baseline.

## Direction (by design)

The viewed bundle is the candidate; the picked bundle is the installed baseline — matching the CLI. So an **added** native plugin (green in the diff) is also what triggers a ❌ verdict (new native code → app-store update); the banner names offenders so the two readings never contradict.

## Files

- **New** `src/services/bundleCompatibility.ts` — pure, unit-tested comparison + verdict util.
- **New** `tests/bundle-compatibility.unit.test.ts` — 14 tests.
- `src/pages/app/[app].bundle.[bundle].dependencies.vue` — status-aware table, verdict banner, `?compare` sync.
- `src/components/bundle/BundleCompareSelect.vue` — exclude deleted bundles; deleted-deployment fallback.
- `messages/en.json` — new i18n keys (source locale; others fall back to `en`).
- Design spec under `docs/superpowers/specs/`.

No backend/schema/API changes.

## Verification

- ✅ 14/14 unit tests
- ✅ Lint: `oxlint src` and `eslint src/**/*.{vue,ts,js}` both clean
- ✅ Typecheck clean (only a pre-existing unrelated `gsap` resolution note)
- ✅ Verified against prod data (`me.wcaleniewolny.test.ionic.vue2`): `1.0.7-c` vs `0.0.0` → 3 changed packages, ❌ Not compatible; deleted `1.0.4-*` bundles excluded from the picker; deploy-history fallback exercised (channel with 4 deleted + 10 live deploys).

## Test plan

- [ ] Open a bundle's Dependencies tab, pick a baseline → verdict banner + colour-coded rows render; dropdown closes on select; URL gains `?compare=`.
- [ ] Load a `?compare=<id>` URL directly → baseline pre-selected, comparison shown.
- [ ] Soft-deleted bundles do not appear in the compare picker (dependencies and manifest tabs).
- [ ] Compare two identical bundles → all unchanged, ✅ Compatible.

## Summary by CodeRabbit

* **New Features**
  * OTA compatibility checking for native dependencies with a verdict banner, per-package compatibility reasons, status summary cards, and a status-aware comparison table.

* **User Experience**
  * Baseline selection is URL-driven, offers richer history choices, and dropdown behavior is more reliable.

* **Documentation**
  * Added a design spec for bundle dependency compatibility.

* **Localization**
  * Updated dependency/compatibility UI copy and added new message keys.

* **Tests**
  * Added unit tests covering compatibility logic and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Cap-go/capgo/pull/2375?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/pages/app/[app].bundle.[bundle].dependencies.vue [hunk 1]
```diff
@@ -128,49 +168,111 @@ async function getVersion() {
...
+  if (!compareId || Number.isNaN(compareId) || compareId === id.value) {
+    replaceCompareParam(null)
+    return
+  }
+  const { data, error } = await supabase
+    .from('app_versions')
+    .select('id, name, created_at, manifest_count, app_id')
+    .eq('app_id', packageId.value)
+    .eq('id', compareId)
+    .maybeSingle()
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply src/pages/app/[app].bundle.[bundle].dependencies.vue:216]: Fixed in 6a3c42d62. `restoreCompareFromQuery` now applies the same eligibility as the picker — `.eq('deleted', false)` and `.not('native_packages', 'is', null)` — so a deep link to a deleted or package-less bundle is scrubbed from the URL instead of comparing against a coerced-empty baseline.

File: src/services/bundleCompatibility.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+  if (!versionsIntersect(candidate.version, baseline.version))
+    reasons.push('version_mismatch')
+
+  const iosChanged = Boolean(
+    candidate.ios_checksum && baseline.ios_checksum && candidate.ios_checksum !== baseline.ios_checksum,
+  )
+  const androidChanged = Boolean(
+    candidate.android_checksum && baseline.android_checksum && candidate.android_checksum !== baseline.android_checksum,
+  )
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply src/services/bundleCompatibility.ts]: Done in e77a2f9bb. Dropped the Boolean() wrappers, using explicit `!= null` guards so iosChanged/androidChanged stay typed as boolean (bare && would infer string | boolean).

**Comment by WcaleNieWolny** [comment]
Addressed the CodeRabbit "outside diff range" finding on `bundleCompatibility.ts` (`statusFor` vs `evaluateReasons`): a same-version package whose native checksum changed was classified `unchanged` while the verdict flagged it incompatible — a contradictory gray-pill-plus-red-reason row. Fixed in 06f36f6ac by deriving status from the computed reasons, so any incompatibility-causing change (version or native checksum) renders as `changed`. Updated 2 tests + added a same-version/same-checksum guard test (15 total, all green). No inline thread existed to resolve (the finding was posted outside the diff range).
